docs/developer/architecture/core/uisettings-service.asciidoc

```diff
@@ -3,14 +3,68 @@
 
 NOTE: The UI settings service is available both server and client side.
 
-=== Server side usage
+=== Overview
+
+UI settings are configurable from the Advanced Settings page in Management and control the behavior of {kib}. `uiSettings` are stored in a `config` saved object and, as such, conform to the same conditions as other saved objects.
+
+There are several ways to configure an advanced setting:
+
+- Through the <<advanced-options, Advanced Settings UI>> (security restrictions apply)
+- Locked via kibana.yml's {kib-repo}blob/{branch}/src/core/server/ui_settings/ui_settings_defaults_client.ts[uiSettings.overrides.<key>]
+- Through the client-side `uiSettings` Service
+- Through the server-side `uiSettings` Service
+
+When a setting is configured as an override in kibana.yml, it will override any other value stored in the config saved object. If an override is misconfigured, it will fail config validation and will prevent Kibana from starting up. The override applies to Kibana as a whole for all spaces and users and the option will be disabled in the Advanced Settings page. We refer to these as "global" overrides.
 
-The program interface to <<advanced-options, UI settings>>.
-It makes it possible for Kibana plugins to extend Kibana UI Settings Management with custom settings.
+The `uiSettings` service is the programming interface to {kib} <<advanced-options, Advanced Settings UI>>. {kib} plugins use the service to extend Kibana UI Settings Management with custom settings for their plugin.
+
+=== Client side usage
+
+On the client, the `uiSettings` service is exposed directly from `core`.
+
+Users can adjust the settings via Management Advanced Settings UI. The client-side {kib-repo}blob/{branch}/docs/development/core/public/kibana-plugin-core-public.iuisettingsclient.md[`uiSettings` client] provides plugins access to the `config` entries stored in elasticsearch.
+
+In the interest of performance, `uiSettings` are cached. Any changes that require cache refreshes (or for whatever other reason) should register an instruction to reload the page when settings are configured in Advanced Settings using the `requiresPageReload` parameter.
```

Comment:
nit: could probably drop the `(or for whatever other reason)` parenthetical

Reply:
😆  good catch! That was a "bloop" from the rough draft!